src/Microsoft.DotNet.XHarness.CLI/Commands/WASM/Browser/WasmTestBrowserCommand.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+
+using Microsoft.DotNet.XHarness.CLI.CommandArguments;
+using Microsoft.DotNet.XHarness.CLI.CommandArguments.Wasm;
+using Microsoft.DotNet.XHarness.Common.CLI;
+using Microsoft.DotNet.XHarness.Common.CLI.Commands;
+using Microsoft.DotNet.XHarness.Common.CLI.CommandArguments;
+using Microsoft.Extensions.Logging;
+
+using OpenQA.Selenium.Chrome;
+using SeleniumLogLevel = OpenQA.Selenium.LogLevel;
+using OpenQA.Selenium;
+
+namespace Microsoft.DotNet.XHarness.CLI.Commands.Wasm
+{
+    internal class WasmTestBrowserCommand : XHarnessCommand
+    {
+        private const string CommandHelp = "Executes tests on WASM using a browser";
+
+        private readonly WasmTestBrowserCommandArguments _arguments = new WasmTestBrowserCommandArguments();
+
+        protected TestCommandArguments TestArguments => _arguments;
+        protected override string CommandUsage { get; } = "wasm test-browser [OPTIONS] -- [BROWSER OPTIONS]";
+        protected override string CommandDescription { get; } = CommandHelp;
+
+        protected override XHarnessCommandArguments Arguments => TestArguments;
+
+        public WasmTestBrowserCommand() : base("test-browser", allowsExtraArgs: true, CommandHelp)
+        {
+        }
+
+        protected override async Task<ExitCode> InvokeInternal(ILogger logger)
+        {
+            var xmlResultsFilePath = Path.Combine(_arguments.OutputDirectory, "testResults.xml");
+            File.Delete(xmlResultsFilePath);
+
+            var logProcessor = new WasmTestMessagesProcessor (xmlResultsFilePath, logger);
+            var runner = new WasmBrowserTestRunner(
+                                _arguments,
+                                PassThroughArguments,
+                                logProcessor.Invoke,
+                                logger);
+
+            using var driver = GetChromeDriver();
+            return await runner.RunTestsWithWebDriver(driver);
+        }
+
+        private IWebDriver GetChromeDriver()
+        {
+            var options = new ChromeOptions();
+            options.SetLoggingPreference(LogType.Browser, SeleniumLogLevel.All);
+            options.AddArguments(new List<string>(_arguments.BrowserArgs)
+            {
+                "--incognito",
+                "--headless"
+            });
+
+            return new ChromeDriver(options);
```

Comment:
This forks off a `chromedriver` instance from $PATH, but if the matching Chrome isn't in a location `chromedriver` can figure out by itself (which will be the case for us on Helix, as the plan is to drop Chrome in the correlation directory not in a system location which `chromedriver` can infer from the registry) then we need a way to tell `chromedriver` where to look for Chrome. The canonical way to set the path to Chrome is via `options.setBinary("/path/to/other/chrome/binary");` and we ought to have a mechanism to pass this to xharness, either via a command-line arg or the environment

Reply:
@directhex can `chromedriver` locate Chrome if it's in the $PATH?

Reply:
You know, I didn't think it would, but apparently yes - https://github.com/chromium/chromium/blob/master/chrome/test/chromedriver/chrome/chrome_finder.cc (and tested locally on MacOS)

Which means if chromedriver and chromium are in the same dir (which they are in my branch) and added to %PATH% then it's fine and works